src/rhel/7.5/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM registry.access.redhat.com/rhel7.5
+
+# Disable repository with undesired packages.
+RUN yum-config-manager --disable rhel-7-server-rt-beta-rpms
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
+# this does not include libraries that we need to compile different projects, we'd like
+# them in a different layer.
+
+# --- Toolchain from RHEL-7 software collection ---
+RUN yum updateinfo \
+    && yum-config-manager --enable rhel-7-server-devtools-rpms \
+    && yum install -y rh-dotnet20-jsoncpp --enablerepo=rhel-7-server-dotnet-rpms \
+        cmake \
+        llvm-toolset-7-clang.x86_64 \
+        llvm-toolset-7-clang-libs.x86_64 \
+        llvm-toolset-7-lldb.x86_64 \
+        llvm-toolset-7-lldb-devel.x86_64 \
+        llvm-toolset-7-llvm.x86_64 \
+        llvm-toolset-7-llvm-libs.x86_64 \
+        wget \
+        which \
+        make \
+    && yum clean all
+
+# Needed by coreclr build to discover llvm-toolset-7 clang, and includes.
+ENV PATH "/opt/rh/llvm-toolset-7/root/usr/bin:${PATH}"
```

Comment:
Consider: declaring both ENV with one instruction/layer.  Be careful of the syntax if you do, it is different - see https://github.com/dotnet/dotnet-docker/blob/master/2.1/sdk/stretch/amd64/Dockerfile#L29

Reply:
I've only made some small formatting changes to this file so far since dleeapho had success with it, but I'll see about this. My concern is `${PATH}`: reading the Docker manual (very) pessimistically makes me think it won't work with `FOO=BAR` syntax:

> https://docs.docker.com/engine/reference/builder/#env
> 
> ...The first form, ENV <key> <value>, will set a single variable to a value. The entire string after the first space will be treated as the <value> - including whitespace characters. **The value will be interpreted for other environment variables, so quote characters will be removed if they are not escaped.**
> 
> **The second form**, ENV <key>=<value> ..., allows for multiple variables to be set at one time. Notice that the second form uses the equals sign (=) in the syntax, while the first form does not. Like command line parsing, quotes and backslashes can be used to include spaces within values.

But it's easy to see if it worked and I have a local build set up, so I'll just try it. It seems like it would be very silly for it not to work.